Course: Data Science Full Course Intensive Training <br>
Name: Richard Kofi Adukonu <br>
Date: 25.02.2022

In [1]:
import requests as rq

In [2]:
url = "https://www.jumia.com.gh/laptops/"


In [3]:
res = rq.get(url)

In [4]:
res # Response code 

<Response [403]>

The HTTP 403 is an HTTP status code meaning access to the requested resource is forbidden. The server understood <br> 
the request, but will not fulfill it. <br>

Learn how to bypass this error here (PYTHON : Problem HTTP error 403 in Python 3 Web Scraping): <br>
"https://youtu.be/b9vcQrIxIYE"

In [5]:
from urllib.request import Request, urlopen

In [6]:
req = Request(url, headers={'User-Agent': 'Mozilla/5.0'})

In [7]:
webpage = urlopen(req).read()
# webpage

In [8]:
from bs4 import BeautifulSoup as bs # import required libraries

In [9]:
soup = bs(webpage, "html.parser")

In [11]:
lap_name = soup.findAll(class_="name")
#lap_name

In [12]:
lap_price = soup.findAll(class_="prc")
#lap_price

In [13]:
Laptop = [i.text.strip() for i in lap_name] # get all job roles
#Laptop

In [14]:
Laptop_prices = [j.text.strip() for j in lap_price] # get all job roles
#Laptop_prices

In [15]:
# After scraping or collecting data, there is the need to store it. 
# For this, we use pandas

import pandas as pd

In [16]:
df1 = pd.DataFrame(Laptop, columns=["Laptop Brand"]) # store the data in a dataframe
#df1

In [17]:
df2 = pd.DataFrame(Laptop_prices, columns=["Laptop Prices"]) # store the data in a dataframe
#df2

In [18]:
frames = [df1.astype(str), df2.astype(str)]

In [19]:
result = pd.concat(frames, axis=1)

In [20]:
result

Laptop Brand   Laptop Prices
0   Lenovo IdeaPad-1-11IGL05 11.6" - 4GB RAM - 128...    GH₵ 1,855.00
1   10.1 inch Netbook PC - 1GBRAM - 8GB HDD - Andr...      GH₵ 750.00
2   N1001 Netbook PC - 10.1" - Dual-Core - 1GB RAM...      GH₵ 719.00
3   Lenovo IdeaPad-1-11IGL05 11.6" - 4GB RAM - 128...    GH₵ 1,855.00
4   Hp Laptop - Intel Celeron Core 2 Duo - 4G RAM ...    GH₵ 1,656.00
5   Lenovo V14-IGL -14.0″ -  Intel N4020 - 4GB DDR...    GH₵ 2,245.00
6   Apple 2020 MacBook Pro - 512GB SSD 8GB RAM -13...    GH₵ 8,900.00
7   Hp Laptop 14 - cf2187nia 10th Generation Intel...    GH₵ 2,799.00
8   10.1 inch Netbook PC - 1GBRAM - 8GB HDD - Andr...      GH₵ 750.00
9   Hp 15 Notebook Laptop -15.6'' - Intel® Celeron...    GH₵ 2,244.00
10  Hp Pavilion x360 14-DW0013DX- Core i3-1005G1- ...    GH₵ 4,200.00
11  Hp Laptop 15.6” - Intel Celeron Core 2 Duo - 4...    GH₵ 1,750.00
12  Hp 15-dw1081wm Intel Pentium Gold 6405U Laptop...    GH₵ 2,399.00
13  Lenovo V15 Intel Core i3 - 10th Generation 15....    GH₵ 3,000.00
14  Hp Notebook 14-DQ1043CL - Intel Core i3-1005G1...    GH₵ 3,060.00
15  Hp 15T-ED000 x360 -15" - Core i7-1065G7+512 SS...    GH₵ 9,800.00
16  Lenovo Ideapad S145-15IGM (81MX) - Intel Celer...    GH₵ 2,189.00
17  Lenovo V15 Intel Core i3 - 10th Generation 15....    GH₵ 2,879.00
18  Hp Notebook 15-da2937nia - 15.6" - 10th Gen Co...    GH₵ 4,393.75
19  Hp Pavilion Gaming -Core i5-10300H -1TB DRIVE ...    GH₵ 5,800.00
20  Lenovo V15 Intel Core i3 - 10th Generation 15....    GH₵ 3,434.00
21  TDD-10.1 Netbook PC - 10.1" - Dual-Core - 1GB ...      GH₵ 855.00
22  Hp Envy x360 convertible 15 Touchscreen - Core...    GH₵ 6,690.00
23  360o Rotating Bluetooth 7 s Backlit Aluminum K...      GH₵ 499.00
24  N1001 Netbook PC - 10.1" - Dual-Core - 1GB RAM...      GH₵ 719.00
25  DELL Vostro 3500 - 11th Generation Intel Core ...    GH₵ 3,607.50
26  DELL Vostro 3000 Series - 11th Gen Intel Core ...    GH₵ 3,295.00
27  Lenovo V14-IGL - Intel Celeron N4020- 1TB HDD ...    GH₵ 2,490.00
28  Foldable Non-slip Laptop N Stand Portable Thin...      GH₵ 129.00
29  Hp Pavilion Gaming -Core i5-10300H - 256SSD- 8...    GH₵ 6,300.00
30  Hp Laptop 14 - cf2187nia 10th Generation Intel...    GH₵ 3,250.00
31  Hp 15-DY1091WM Laptop – Core™ i3 1.20GHz – 10t...    GH₵ 3,399.00
32  Hp Pavilion x360 14-DH2051WM - 10th Gen- 8GB R...  GH₵ 100,000.00
33  Hp 250 G7 - 15.6" - Intel Celeron N4000 - 1TB ...    GH₵ 2,489.00
34  Hp Laptop 15.6” - Intel Celeron Core 2 Duo -50...    GH₵ 1,820.00
35  DELL Vostro 3000 Series - 11th Gen Intel Core ...    GH₵ 3,055.00
36  Hp Notebook 15 - 15.6" - 10th Gen Core i5 Proc...    GH₵ 3,885.00
37  Hp 15-da0021ds Intel Pentium Gold 5405U Laptop...    GH₵ 2,950.00
38  10.1 inch  Netbook PC -1GBRAM - 8GBROM - Andro...      GH₵ 779.00
39  Hp Pavilion 15-eg0073cl 15.6″ Touchscreen Lapt...    GH₵ 6,990.00
40  Apple MacBook Pro M1- 256GB-HDD-8GB-RAM-13.3-S...    GH₵ 9,000.00
41  Apple MacBook Pro - 256GB SSD 8GB RAM -13.3 -S...    GH₵ 9,000.00
42  Art Graphics Tablet Drawing Pen Wireless Digit...        GH₵ 6.00
43                                                                   
44                                                                   
45                                                                   
46                                                                   
47                                                                   
48                                                                   
49                                                                   
50

In [21]:
# save our data in a csv file
result.to_csv("Laptops_and_Prices_from_Jumia.csv", encoding="utf-8-sig")